### ISSUE TITLE
PostCSS + Sass

### ISSUE THREAD

**Issue by philipnilsson** [andywer/webpack-blocks#151]
What is the approach for using both the PostCSS and SASS blocks? 

I've modified the PostCSS plugin to set 'text/x-sass' as the filetype instead of 'text/css' and everything works, but there seems to be no way of specifying this in the plugin.

Out of the box PostCSS just applies to to `.css` files of which I have none, and do not seem to be picked up at all after running the SASS compiler.

**Comment by andywer** [comment]
Yeah, that seems to be a missing feature. Thanks for sharing, @philipnilsson

There is also an end-to-end test testing PostCSS & SASS, but as it turns out it does a poor job testing the interaction of those two... 🙃

**Comment by philipnilsson** [comment]
Alright. My workaround for anyone encountering the same issue. 

```
const postcssSASS = options => context => {
  const config = postcss(options)(context);
  config.module.loaders[0].test =
    context.fileType('text/x-sass');
  return config;
};
```

I'd be happy to PR something to make this more easily configurable, but it's a bit annoying that it seems hard to do something API-compatible with the `extractText` block which has this feature, at least without breaking backwards compatibility in `postcss`.

**Comment by andywer** [comment]
I think there is an easy way:

The `postcss` block accepts an options object as 2nd parameter. Let's just add a `fileType` property to these options that defaults to `text/css`.

Only downside I see: Setting a custom file type looks slightly differently for `postcss` and `sass`.

**Comment by sapegin** [comment]
In my understanding `css`, `sass` and `postcss` are too close to each other, so maybe we need some kind of `style` block would make sense. Not sore what could be a good API though.

**Comment by sapegin** [comment]
Or it could be added to a `css` block like this:

```js
css.modules({
  fileType: 'text/x-sass',
  loaders: [
      'postcss-loader',
      {
          loader: 'sass-loader',
           options: {
               includePaths: [path.join(__dirname, 'some/path')],
           },
       },
  ]
}),
```

So may use `css` block to define `sass` and `postcss` blocks instead of copy pasting. Also [PostCSS config files](https://github.com/michael-ciniawsky/postcss-load-config) seems to be a recommended way to configure PostCSS now, so it make sense not to support custom options by default (so you could get rid of most of the code in `postcss` block, especially `addLoaderOptionsPlugin`). (But actually supporting custom options [shouldn’t](https://github.com/postcss/postcss-loader/issues/193#issuecomment-291563828) be difficult now.)

**Comment by andywer** [comment]
Thanks for sharing! But I think that would prevent any composition which is the primary reason why we have webpack-blocks ^^

I have thought about something like this recently:

```js
createConfig([
  match('*.css', [
    css(),
    css.modules(),
    postcss()
  ])
])
```

But composing fragments of a loader definition is not yet possible. Still thinking if/how this could be achieved...
What do you think?

**Comment by sapegin** [comment]
It’s very non-obvious now that several blocks will be merged together and looks like there’s no consistent way to pass styles type.

You proposal is interesting too. And you can implement my in any case to simplify code of styles related blocks ;-)

**Comment by andywer** [comment]
God, that issue is old 🙈
`match()` has already been implemented some time ago: [v1.0.0-beta](https://github.com/andywer/webpack-blocks/releases/tag/v1.0.0-beta)

Use that 😉
